ossfuzz/aspell_fuzzer.cpp

```diff
@@ -0,0 +1,152 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <aspell.h>
+
+#define FUZZ_LANG_CODE_LEN              5
+#define FUZZ_ENCODING_LEN               1
+#define NUM_COMMAND_BYTES               (FUZZ_LANG_CODE_LEN + \
+                                         FUZZ_ENCODING_LEN)
+
+static int enable_diags;
+
+#define FUZZ_DEBUG(FMT, ...)                                                  \
+        if (enable_diags) {                                                   \
+          fprintf(stderr, FMT, ##__VA_ARGS__);                                \
+          fprintf(stderr, "\n");                                              \
+        }
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+  AspellCanHaveError *possible_err = NULL;
+  AspellSpeller *spell_checker = NULL;
+  AspellConfig *spell_config = NULL;
+  AspellDocumentChecker *doc_checker = NULL;
+  AspellCanHaveError *doc_err = NULL;
+  AspellToken token;
+  const char *data_str = reinterpret_cast<const char *>(data);
+
+  char language_code[FUZZ_LANG_CODE_LEN + 1];
+  const char *encoding;
+
+  // Enable or disable diagnostics based on the FUZZ_VERBOSE environment flag.
+  enable_diags = (getenv("FUZZ_VERBOSE") != NULL);
+
+  if (size < NUM_COMMAND_BYTES) {
+    goto EXIT_LABEL;
+  }
+
+  // Copy the first five bytes as a language code and null-terminate it.
+  memcpy(language_code, data, FUZZ_LANG_CODE_LEN);
+  language_code[FUZZ_LANG_CODE_LEN] = 0;
+
+  data += FUZZ_LANG_CODE_LEN;
+
+  // Determine the encoding.
+  switch (data[0] % 4) {
+    case 0:
```

Comment:
I _think_ this should be `'0'` as it looks like you are testing for a digit 0 and not the char value of zero.

Reply:
Nope, my aim here is to treat the unsigned char as a number between 0 and 255, and then calculate modulo 4, which will give 0, 1, 2 or 3 (which corresponds to the switch cases, hence there being no default because the cases are exhaustive)

Reply:
Sorry, I missed the `% 4` part.

I any case `iso-8859-*` is not a valid encoding string, (the '*' was not meant to be taken literary in the manual).  Just use `iso-8859-1` for now.

Reply:
Ok, fixed.